services/src/main/java/org/keycloak/authentication/requiredactions/UpdatePassword.java

```diff
@@ -205,12 +239,62 @@ public int getMaxAuthAge() {
             return MaxAuthAgePasswordPolicyProviderFactory.DEFAULT_MAX_AUTH_AGE;
         }
 
-        int maxAge = session.getContext().getRealm().getPasswordPolicy().getMaxAuthAge();
-        if (maxAge < 0) {
-            // passwordPolicy is not present fallback to default maxAuthAge
-            return MaxAuthAgePasswordPolicyProviderFactory.DEFAULT_MAX_AUTH_AGE;
+        // try password policy
+        KeycloakContext keycloakContext = session.getContext();
+        RealmModel realm = keycloakContext.getRealm();
+        int maxAge = realm.getPasswordPolicy().getMaxAuthAge();
+        if (maxAge >= 0) {
+            return maxAge;
+        }
+
+        // try required action config
+        AuthenticationSessionModel authSession = keycloakContext.getAuthenticationSession();
+        if (authSession != null) {
+
+            // we need to figure out the alias for the current required action
+            String providerId = authSession.getClientNote(Constants.KC_ACTION);
+            RequiredActionProviderModel requiredAction = realm.getRequiredActionProvidersStream() //
```

Comment:
Nitpick: I can imagine it can be useful to have some util method (maybe in the `RequiredActionsHelper` as suggested in other comments or in `KeycloakModelUtils`) like:
```
RequiredActionConfigModel getRequiredActionConfig(RealmModel realm, String requiredActionProviderId)
```
as I suppose the similar code will apply always when retrieving configurations from required action